core/src/test/java/org/elasticsearch/search/aggregations/AggregationsTests.java

```diff
@@ -175,11 +178,43 @@ public void testAllAggsAreBeingTested() {
     }
 
     public void testFromXContent() throws IOException {
+        parseAndAssert(false);
+    }
+
+    public void testFromXContentWithRandomFields() throws IOException {
+        parseAndAssert(true);
+    }
+
+    private void parseAndAssert(boolean addRandomFields) throws IOException {
```

Comment:
nit: can we add a small comment about what this method is doing? No need to go in details but just the overall logic and why we do this.